### ISSUE TITLE
zip failed when process.exit() after await archive.finalize()

### ISSUE THREAD

**Issue by xkn1ght** [archiverjs/node-archiver#555]
### Discovered a likely bug.

when I want to await the archive.finalize() and exit my program, my zip file would be broken when unzipping it.

here is my [case](https://github.com/xkn1ght/finalize-failed)

my node version is v14.17.4

please tell me if i used it in a wrong way

**Comment by xkn1ght** [comment]
core code here
```ts
import archiver from 'archiver';
import fs from 'fs';

async function zip(from: string, to: string,): Promise<void> {
    const ZIP_LEVEL = 9;
    const archive = archiver('zip', { zlib: { level: ZIP_LEVEL } });
    const stream = fs.createWriteStream(to);
    archive
        .directory(from, false)
        .on('error', err => {
            throw err;
        })
        .pipe(stream);

    await archive.finalize();
}

async function main(){
    await zip('./test','out.zip');
    process.exit();
}

main();
```

**Comment by jasrusable** [comment]
Try resolve your promise only after the stream has closed, like this example: https://github.com/archiverjs/node-archiver/issues/494#issuecomment-828479906

**Comment by xkn1ght** [comment]
> Try resolve your promise only after the stream has closed, like this example: [#494 (comment)](https://github.com/archiverjs/node-archiver/issues/494#issuecomment-828479906)
thx~
